lib/core/engine/iteration.js

```diff
@@ -217,7 +227,31 @@ class Iteration {
                 if (!geckoProfile.meta) {
                   geckoProfile.meta = {};
                 }
-                geckoProfile.meta.visualMetrics = result[i].visualMetrics;
+
+                // Here we are calculating the unix timestamp of the first frame after orange screen
+                // by adding the time to first frame, time to orange screen to the unix timestamp of
+                // video recording start time
+                const firstFrameStartTime =
+                  result[i].recordingStartTime +
+                  videoMetrics.videoRecordingStart +
+                  result[i].timeToFirstFrame;
+
+                videoMetrics.visualMetrics.VisualProgress = util.adjustVisualProgressTimestamps(
```

Comment:
Same thing here: consider `for (var of ...)` so that adding/tweaking new index calculations is obvious.